src/js/options.js

```diff
@@ -273,21 +276,48 @@ function refreshOriginCache() {
 
 /**
  * Gets array of encountered origins.
+ *
  * @param filterText {String} Text to filter origins with.
+ * @param type_filter {String} Type (user-controlled, DNT-compliant) to filter
+ *   origins by.
+ * @param status_filter {String} Status (blocked, cookieblocked, allowed) to
+ *   filter origins by.
+ *
  * @return {Array}
  */
-function getOriginsArray(filterText) {
+function getOriginsArray(filterText, type_filter, status_filter) {
   // Make sure filterText is lower case for case-insensitive matching.
   if (filterText) {
     filterText = filterText.toLowerCase();
   } else {
     filterText = "";
   }
 
-  // Include only origins containing given filter text.
+  // Include only origins that pass given filters.
   function containsFilterText(origin) {
+    const value = originCache[origin];
+
+    if (type_filter) {
+      if (type_filter == "user") {
+        if (!value.startsWith("user")) {
+          return false;
+        }
+      } else {
+        if (value != type_filter) {
+          return false;
+        }
+      }
+    }
+
+    if (status_filter) {
+      if (status_filter != value.replace("user_", "")) {
+        return false;
+      }
+    }
```

Comment:
[Style] Mix of `==` and `===` in some of these functions; do we have any preference for this?

Reply:
No strong preference ... I'll kick this particular can down the road for now.